yew/src/virtual_dom/vtag.rs

```diff
@@ -110,6 +118,84 @@ impl Clone for VTag {
     }
 }
 
+/// HTML output for a VTag is not necessarily deterministic due to the
+/// serialization of props which do not have a particular ordering.
+#[cfg(feature = "ssr")]
+impl TryFrom<VTag> for Html {
+    type Error = HtmlStringifyError;
+
+    fn try_from(value: VTag) -> Result<Html, HtmlStringifyError> {
+        let mut parts: Vec<String> = Vec::new();
+        let tag_name = htmlescape::encode_minimal(&value.tag).to_lowercase();
+
+        for c in tag_name.chars() {
+            let is_alnum = (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9');
+            if !is_alnum && c != '-' && c != '_' && c != ':' {
+                return Err(HtmlStringifyError::InvalidTagName(tag_name));
+            }
```

Comment:
If we want to allow custom elements this character range is too narrow. The characters `-`, `_`, `:` aren't allowed as part of a 'vanilla' tag name<sup>[[1]](https://html.spec.whatwg.org/multipage/syntax.html#syntax-tag-name)</sup> so we definitely want to expand this.
Here's the [HTML Standard on custom element names](https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name) for reference.

Could we also extract this into a function? We need to have this function for #1269 anyway and the same check appears again further down.

Reply:
@siku2 I agree that we can extract this into a function. Custom tag names sound good. We probably want a function which works as a small parser.

Reply:
@siku2 Reading into that a bit, I am curious what the exact use case for this is. From the resource:

> Apart from these restrictions, a large variety of names is allowed, to give maximum flexibility for use cases like <math-α> or <emotion-😍>.

This seems a bit crazy to me and not like a rabbit hole we'd want to go down without a good reason.

Reply:
I'm surprised you feel that way... As I see it, the reason why we should support this is to make Yew as flexible as possible. Custom elements are well-supported and there are quite a few libraries out there providing them so I don't see why we wouldn't want to support them.
Yew can (and should) be opinionated, of course, but I feel like it shouldn't make certain things impossible.

Reply:
Perhaps we should even remove the check here entirely and instead assume that the tag name is valid.
The actual check could then be performed while creating the tag like proposed in #1269.

Reply:
I went ahead and started implementing the logic surrounding this, although the code is quite verbose and probably could be split into its own crate. (it's in this PR for now)